src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -3342,6 +3502,111 @@ void ApplicationManagerImpl::RemoveAppFromTTSGlobalPropertiesList(
   }
   tts_global_properties_app_list_lock_.Release();
 }
+#ifdef SDL_REMOTE_CONTROL
+void ApplicationManagerImpl::CreatePhoneCallAppList() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  DataAccessor<ApplicationSet> accessor = applications();
+
+  ApplicationManagerImpl::ApplictionSetIt it = accessor.GetData().begin();
+  ApplicationManagerImpl::ApplictionSetIt itEnd = accessor.GetData().end();
+
+  using namespace mobile_apis::HMILevel;
+  using namespace helpers;
+  for (; it != itEnd; ++it) {
+    if (Compare<eType, EQ, ONE>((*it)->hmi_level(), HMI_FULL, HMI_LIMITED)) {
+      // back up app state
+      on_phone_call_app_list_.insert(
+          std::pair<uint32_t, AppState>((*it)->app_id(),
+                                        AppState((*it)->hmi_level(),
+                                                 (*it)->audio_streaming_state(),
+                                                 (*it)->system_context())));
+
+      ChangeAppsHMILevel((*it)->app_id(),
+                         (*it)->is_navi() ? HMI_LIMITED : HMI_BACKGROUND);
+
+      // app state during phone call
+      (*it)->set_audio_streaming_state(
+          mobile_api::AudioStreamingState::NOT_AUDIBLE);
+      (*it)->set_system_context(mobile_api::SystemContext::SYSCTXT_MAIN);
+      MessageHelper::SendHMIStatusNotification(*(*it), *this);
+    }
+  }
+}
+
+void ApplicationManagerImpl::ResetPhoneCallAppList() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  std::map<uint32_t, AppState>::iterator it = on_phone_call_app_list_.begin();
+  std::map<uint32_t, AppState>::iterator it_end = on_phone_call_app_list_.end();
+  for (; it != it_end; ++it) {
+    ApplicationSharedPtr app = application(it->first);
+    if (app) {
+      ChangeAppsHMILevel(app->app_id(), it->second.hmi_level);
+
+      app->set_audio_streaming_state(it->second.audio_streaming_state);
+      app->set_system_context(it->second.system_context);
+      MessageHelper::SendHMIStatusNotification(*app, *this);
+    }
+  }
+
+  on_phone_call_app_list_.clear();
+}
+
+void ApplicationManagerImpl::ChangeAppsHMILevel(
```

Comment:
@LevchenkoS please try to use state_ctrl for that purpose. If it is RevSDL code, create TODO and task in JIRA to make in in state CTRL